typing/typecore.ml

```diff
@@ -3112,12 +3225,14 @@ let type_omitted_parameters expected_mode env ty_ret mode_ret args =
                  (fun (marg, exp) ->
                     submode ~loc:exp.exp_loc ~env ~reason:Other
                       marg (mode_partial_application expected_mode);
-                    Value_mode.regional_to_local_alloc marg)
+                    Value.regional_to_local_alloc marg)
                  open_args
              in
              let closed_args = new_closed_args @ closed_args in
              let open_args = [] in
-             let mode_closure = Alloc_mode.join (mode_fun :: closed_args) in
+             let m0 = List.map Alloc.close_over closed_args in
+             let m1 = Alloc.partial_apply mode_fun in
+             let mode_closure, _ = Alloc.newvar_above (Alloc.join (m1 :: m0)) in
```

Comment:
I can't remember why I did this, but it might be that `Alloc.join` sometimes gives the _least_ upper bound, instead of just upper bound, and therefore sometimes it's not pushed to `local` when it should.

In the new solver, `join` gives a "left" mode, while "register_allocation_mode" would require a "right" mode (because you want to say `local <= m`. So this `newvar_above` is needed. There are several occurrences in the new solver PR where I have to make new vars to type check. Unfortunately I didn't backport each of them here, leaving potential BUGs.